Shared/Action/TelemetryAction.swift

```diff
@@ -61,6 +62,32 @@ enum ExtraKey: String {
     case fxauid, itemid, error
 }
 
+class GleanActionHandler: ActionHandler {
+    private let glean = Glean.shared
+    
+    private var disposeBag = DisposeBag()
+    
+    init() {
+        UserDefaultStore.shared.recordUsageData.subscribe(
+            onNext: { uploadEnabled in
+                self.glean.setUploadEnabled(uploadEnabled)
```

Comment:
Does this guarantee that `setUploadEnabled` gets called before `glean.initialize()`? Can we add a comment about that and about why it's required?

Reply:
Yes, according to the docs calling `subscribe` invokes the preference which should trigger the `onNext` block.

I'll add a comment to make it clear that `setUploadEnabled` needs to be called prior to `initialize`

Reply:
Both `Glean.shared` and `UserDefaultStore.shared` should be passed in as constructor variables.

Reply:
> Does this guarantee that setUploadEnabled gets called before glean.initialize()? Can we add a comment about that and about why it's required?

We could convince ourselves by writing a test in `UserDefaults+Spec.swift`.